### PR TITLE
test(core): add unit tests for normalizeHeaders and createFetchWithInit

### PR DESCRIPTION
**Pull request by k61b**
Add unit tests for `normalizeHeaders` and `createFetchWithInit` in `packages/core/src/shared/transport.ts`.

## Motivation and Context

These utility functions are used by both `SSEClientTransport` and `StreamableHTTPClientTransport` but had no dedicated unit tests. Adding coverage helps catch regressions and documents expected behavior.

## How Has This Been Tested?

```bash
pnpm -C packages/core test
# 441 tests passed
```

## Breaking Changes

None. Test-only change.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

(This is a test-only change)

## Checklist

- [x] I have read the [MCP Documentation](https://modelcontextprotocol.io)
- [x] My code follows the repository's style guidelines
- [x] New and existing tests pass locally
- [ ] I have added appropriate error handling
- [ ] I have added or updated documentation as needed

## Additional context

Tests cover:

- `normalizeHeaders`: undefined, Headers instance, array of tuples, plain objects
- `createFetchWithInit`: base init passthrough, option merging, header merging precedence

### CODE REVIEW

File: packages/core/test/shared/transport.test.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        await wrappedFetch('https://example.com', { method: 'PUT' });
+
+        expect(mockFetch).toHaveBeenCalledWith(
+            'https://example.com',
+            expect.objectContaining({
+                method: 'PUT',
+                credentials: 'include'
+            })
+        );
+    });
```
[the remark this answers is not in the document]
Comment (k61b) [reply packages/core/test/shared/transport.test.ts:89]: Added test case for empty init object.

File: packages/core/test/shared/transport.test.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
+import { createFetchWithInit, type FetchLike, normalizeHeaders } from '../../src/shared/transport.js';
+
+describe('normalizeHeaders', () => {
+    test('returns empty object for undefined', () => {
```
[the remark this answers is not in the document]
Comment (k61b) [reply packages/core/test/shared/transport.test.ts:4]: Both `test` and `it` are used in this codebase. For example, `stdio.test.ts` uses `test` while `uriTemplate.test.ts` uses `it`. Since there's no enforced standard, I'm keeping `test` for consistency within this file.

File: packages/core/test/shared/transport.test.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        await wrappedFetch('https://example.com', { method: 'POST' });
+
+        expect(mockFetch).toHaveBeenCalledWith(
+            'https://example.com',
+            expect.objectContaining({
+                method: 'POST',
+                headers: { 'x-base': 'base-value' }
+            })
+        );
+    });
```
[the remark this answers is not in the document]
Comment (k61b) [reply packages/core/test/shared/transport.test.ts:130]: Added test case verifying Headers instance is passed through when init has no headers.

File: packages/core/test/shared/transport.test.ts [hunk 4]
```diff
@@ -0,0 +1,142 @@
+import { createFetchWithInit, type FetchLike, normalizeHeaders } from '../../src/shared/transport.js';
```
[the remark this answers is not in the document]
Comment (k61b) [reply packages/core/test/shared/transport.test.ts:1]: The project uses `globals: true` in the vitest configuration, which makes `vi`, `describe`, `test`, and `expect` available globally without explicit imports. All tests pass without this import.

File: packages/core/test/shared/transport.test.ts [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        await wrappedFetch('https://example.com');
+
+        expect(mockFetch).toHaveBeenCalledWith(
+            'https://example.com',
+            expect.objectContaining({
+                method: 'POST',
+                credentials: 'include'
+            })
+        );
+    });
```
[the remark this answers is not in the document]
Comment (k61b) [reply packages/core/test/shared/transport.test.ts:70]: This adds assertion noise without meaningful value. If the wrapper called fetch multiple times, the `toHaveBeenCalledWith` assertions would fail due to ambiguous calls. The current assertions are sufficient.

File: packages/core/test/shared/transport.test.ts [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+        const mockFetch: FetchLike = vi.fn();
+        const baseInit: RequestInit = { method: 'GET' };
+
+        const wrappedFetch = createFetchWithInit(mockFetch, baseInit);
+        const url = new URL('https://example.com/path');
+        await wrappedFetch(url);
+
+        expect(mockFetch).toHaveBeenCalledWith(url, expect.objectContaining({ method: 'GET' }));
+    });
+});
```
[the remark this answers is not in the document]
Comment (k61b) [reply packages/core/test/shared/transport.test.ts:182]: These tests focus on verifying the _argument merging_ behavior, not the fetch pass-through. The wrapper is a simple `return baseFetch(url, mergedInit)` - testing that it returns what fetch returns would be testing JavaScript itself.

File: packages/core/test/shared/transport.test.ts [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+            'https://example.com',
+            expect.objectContaining({
+                headers: {
+                    'x-base': 'base-value',
+                    'x-call': 'call-value',
+                    'x-shared': 'call'
+                }
+            })
+        );
+    });
```
[the remark this answers is not in the document]
Comment (k61b) [reply packages/core/test/shared/transport.test.ts:112]: The `normalizeHeaders` function is tested separately and handles all formats. `createFetchWithInit` delegates to `normalizeHeaders`, so testing every permutation would be redundant. The header merging test already verifies the integration works.

[the remark this answers is not in the document]
Comment (k61b) [reply packages/core/test/shared/transport.test.ts:112]: This case is implicitly covered by the "merges headers from both base and call init" test where base headers exist. The code path for "init has headers" is already exercised. Adding a separate test for this specific scenario adds marginal value.

File: packages/core/test/shared/transport.test.ts [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+        });
+    });
+
+    test('returns a shallow copy for plain objects', () => {
+        const headers = { 'x-foo': 'bar' };
+        const result = normalizeHeaders(headers);
+        expect(result).not.toBe(headers);
+        expect(result).toEqual(headers);
+    });
+});
```
[the remark this answers is not in the document]
Comment (k61b) [reply packages/core/test/shared/transport.test.ts:44]: We test `normalizeHeaders(undefined) → {}` which is the primary edge case. Empty `Headers()`, `[]`, and `{}` all produce `{}` through the same code paths. Over-testing these permutations adds maintenance burden without catching real bugs.

**Comment by felixweinberger** [comment]
@claude review

**Review by felixweinberger — APPROVED** [review]
Seems fine to add, thanks!
